pkg/agent/agentclient/stream.go

```diff
@@ -205,7 +239,13 @@ func (c *RedialableAgentClient) tryConnect() error {
 		return err
 	}
 
-	c.stream, err = agent.NewAgentServiceClient(c.conn).Connect(context.Background())
+	ctx := context.Background()
+	if c.saTokenFile != "" {
+		if ctx, err = c.initializeAuthContext(ctx); err != nil {
```

Comment:
We can remove line 242 and 243

Reply:
I removed validation from inside initializeAuthContext method, so will keep it here